dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceCache.java

```diff
@@ -0,0 +1,220 @@
+ *        from cache Our services methods which generate the velocity files will handle the filling
+ *        and removing of The cache. If something is not in cache though the DotResourceLoader will
+ *        be called.
+ */
+public class DotResourceCache implements ResourceCache, Cachable {
+
+
+    private DotCacheAdministrator cache;
+
+    private String primaryGroup = "VelocityCache";
+    private String macroCacheGroup = "VelocityMacroCache";
+    // region's name for the cache
+    private String[] groupNames = {primaryGroup, macroCacheGroup};
+    private static final String MACRO_PREFIX = "MACRO_PREFIX";
+    private final Set<String> ignoreGlobalVM;
+
+
+
+    public DotResourceCache() {
+        cache = CacheLocator.getCacheAdministrator();
+        String files = System.getProperty(RuntimeConstants.VM_LIBRARY);
+        Set<String> holder = new HashSet<>();
+        for (String file : files.split(",")) {
+            if (file != null) {
+                // System.out.println("FILE: " +file);
+                holder.add(file.trim());
+            }
+        }
+        ignoreGlobalVM = ImmutableSet.copyOf(holder);
+    }
+
+    public String[] getMacro(String name) {
+
+
+        String[] rw = null;
+        try {
+            rw = (String[]) cache.get(MACRO_PREFIX + name, macroCacheGroup);
+        } catch (DotCacheException e) {
+            Logger.debug(this, "Cache Entry not found", e);
+        }
+        return rw;
+
+    }
+
+    public void putMacro(String name, String content) {
+        if (name == null || content == null) {
+            Logger.warn(this.getClass(), "Cannot add a null macro to cache:" + name + " / " + content);
+            return;
+        }
+        String[] rw = {name, content};
+        cache.put(MACRO_PREFIX + name, rw, macroCacheGroup);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.apache.velocity.runtime.resource.ResourceCache#get(java.lang.Object)
+     */
+    public Resource get(Object resourceKey) {
+
+        String cleanedResourceKey = cleanKey(resourceKey.toString());
+        String group = primaryGroup;
+
+        Resource resource = null;
+        String key = group + cleanedResourceKey;
+        try {
+            resource = (Resource) cache.get(key, group);
+        } catch (DotCacheException e) {
+            Logger.debug(this, "Cache Entry not found", e);
+        }
+        return resource != null ? resource : null;
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.apache.velocity.runtime.resource.ResourceCache#initialize(org.apache.velocity.runtime.
+     * RuntimeServices)
+     */
+    public void initialize(RuntimeServices rs) {
+        cache = CacheLocator.getCacheAdministrator();
+    }
+
+    public void addMiss(Object resourceKey) {
+        Logger.info(this.getClass(), "velocityMiss:" + resourceKey);
+    }
+
+    public boolean isMiss(Object resourceKey) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "resourceKey". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)